runtime/calamari/src/zenlink.rs

```diff
@@ -0,0 +1,256 @@
+    LocalAssetHandler, ZenlinkMultiAssets, LOCAL, NATIVE,
+};
+use zenlink_stable_amm::traits::{StablePoolLpCurrencyIdGenerate, ValidateCurrency};
+
+parameter_types! {
+    pub const ZenlinkPalletId: PalletId = PalletId(*b"/zenlink");
+    pub const GetExchangeFee: (u32, u32) = (3, 1000);   // 0.3%
+    pub SelfParaId: u32 = ParachainInfo::parachain_id().into();
+    pub MantaNativeAssetId: CalamariAssetId = 1;
+    pub ZenlinkNativeAssetId: u64 = 0;
+
+    pub const AnyNetwork: NetworkId = NetworkId::Any;
+    // Not allowed parachain token transfer on zenlink protocol
+    pub ZenlinkRegistedParaChains: Vec<(MultiLocation, u128)> = vec![];
+}
+
+type MultiAssets = ZenlinkMultiAssets<ZenlinkProtocol, Balances, LocalAssetAdaptor>;
+
+pub type ZenlinkLocationToAccountId = (
+    // Sibling parachain origins convert to AccountId via the `ParaId::into`.
+    SiblingParachainConvertsVia<Sibling, AccountId>,
+    // Straight up local `AccountId32` origins just alias directly to `AccountId`.
+    AccountId32Aliases<RelayNetwork, AccountId>,
+);
+
+impl zenlink_protocol::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type MultiAssetsHandler = MultiAssets;
+    type PalletId = ZenlinkPalletId;
+    type SelfParaId = SelfParaId;
+    type TargetChains = ZenlinkRegistedParaChains;
+    type AssetId = ZenlinkAssetId;
+    type AssetIdConverter = MantaAssetIdConverter;
+    type LpGenerate = AssetManagerLpGenerate;
+    type AccountIdConverter = ZenlinkLocationToAccountId;
+    type XcmExecutor = ();
+    type WeightInfo = ();
+}
+
+parameter_types! {
+    pub const StringLimit: u32 = 50;
+    pub const StableAmmPalletId: PalletId = PalletId(*b"mt/stamm");
+}
+
+impl zenlink_stable_amm::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type CurrencyId = CalamariAssetId;
+    type MultiCurrency = MantaCurrencies;
+    type PoolId = CalamariAssetId;
+    type TimeProvider = Timestamp;
+    type EnsurePoolAsset = StableAmmVerifyPoolAsset;
+    type LpGenerate = PoolLpGenerate;
+    type PoolCurrencySymbolLimit = StringLimit;
+    type PalletId = StableAmmPalletId;
+    type WeightInfo = ();
+}
+
+impl zenlink_swap_router::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type StablePoolId = CalamariAssetId;
+    type Balance = u128;
+    type StableCurrencyId = CalamariAssetId;
+    type NormalCurrencyId = ZenlinkAssetId;
+    type NormalAmm = ZenlinkProtocol;
+    type StableAMM = ZenlinkStableAMM;
+    type WeightInfo = ();
+}
+
+pub struct StableAmmVerifyPoolAsset;
+
+impl ValidateCurrency<CalamariAssetId> for StableAmmVerifyPoolAsset {
+    fn validate_pooled_currency(_currencies: &[CalamariAssetId]) -> bool {
+        true
+    }
+
+    fn validate_pool_lp_currency(_currency_id: CalamariAssetId) -> bool {
+        if <MantaCurrencies as MultiCurrency<<Runtime as frame_system::Config>::AccountId>>::total_issuance(_currency_id) > 0 {
```

Comment:
Just feel a little confused, why bigger than 0 should return false?
And this part of implementation can be simplified as like 
```rust
<MantaCurrencies as MultiCurrency<<Runtime as frame_system::Config>::AccountId>>::total_issuance(_currency_id) <= 0
```